agent/src/softwarecontaineragent.cpp

```diff
@@ -192,17 +205,11 @@ ContainerID SoftwareContainerAgent::findSuitableId()
 
 SoftwareContainerAgent::SoftwareContainerPtr SoftwareContainerAgent::makeSoftwareContainer(const ContainerID containerID)
 {
-    std::string bridgeIp = m_config->getStringValue(ConfigDefinition::SC_GROUP,
-                                                    ConfigDefinition::SC_BRIDGE_IP_KEY);
-    int netmaskBits = m_config->getIntValue(ConfigDefinition::SC_GROUP,
-                                            ConfigDefinition::SC_BRIDGE_NETMASK_BITS_KEY);
-
     log_debug() << "Created container with ID :" << containerID;
 
     auto container = SoftwareContainerPtr(new SoftwareContainer(m_softwarecontainerWorkspace,
                                                                 containerID,
-                                                                bridgeIp,
-                                                                netmaskBits));
+                                                                m_containerConfig));
```

Comment:
Yes, this is not nice at all. I think it's better for the whole design to perhaps have two separate objects in the agent that each are responsible for static and dynamic configs respectively, and "re-pack" it into an object sent to SC on creation, and do all necessary copying at that point. Will rework this.